pkg/controller/controller.go

```diff
@@ -443,20 +447,22 @@ func (p *csiProvisioner) Delete(volume *v1.PersistentVolume) error {
 	// get secrets if StorageClass specifies it
 	storageClassName := volume.Spec.StorageClassName
 	if len(storageClassName) != 0 {
-		if storageClass, err := p.client.StorageV1().StorageClasses().Get(storageClassName, metav1.GetOptions{}); err == nil {
-			// Resolve provision secret credentials.
-			// No PVC is provided when resolving provision/delete secret names, since the PVC may or may not exist at delete time.
-			provisionerSecretRef, err := getSecretReference(provisionerSecretNameKey, provisionerSecretNamespaceKey, storageClass.Parameters, volume.Name, nil)
-			if err != nil {
-				return err
-			}
-			credentials, err := getCredentials(p.client, provisionerSecretRef)
-			if err != nil {
-				return err
-			}
-			req.ControllerDeleteSecrets = credentials
+		storageClass, err := p.client.StorageV1beta1().StorageClasses().Get(storageClassName, metav1.GetOptions{})
```

Comment:
I'm not sure about this change. The test needed creating `StorageClass` objects and adding them to fake clientSet. But since I couldn't figure out any way to do that with v1 API, I ended up used v1beta1 API. Some help in doing this properly without any side effects would be great.